### PR TITLE
Search and employ

### LINKED ISSUE #16: Add logic for request users to join company
Company admins should have form where they can search employees (users) by
name, email, personal_number
Admins can search by personal id, but that record should not be displayed in any results.
That is sensitive data and it has to be entered fully.

For other he can search by partly entered name or email.

From this page, company admin can request from employ to join company, but employee has to accept it.

### PR DESCRIPTION
**Pull request by Baxxx**
Closes #16

### CODE REVIEW

File: app/assets/javascripts/users.js [hunk 1]
```diff
@@ -41,11 +41,16 @@ function search() {
           out += '  </tr> ';
 
         data.users.forEach(function(user){
+          console.log(user)
```
Comment (IvanPavlovic92) [inline app/assets/javascripts/users.js]: remove console.log before commit

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -9,21 +9,32 @@
...
+                </li>
 
+                <li class="dropdown"> <!-- COMPANY OPTIONS DROPDOWN -->
+                  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
+                    Company Options
+                    <span class="caret"></span>
+                  </a>
+                  <ul class="dropdown-menu">
+                    <li><%= link_to 'View ' + current_user.current_company.name, company_path(current_user.current_company) %> </li>
+                    <li><%= link_to 'Search for employees', user_search_path %> </li>
```
Comment (IvanPavlovic92) [inline app/views/layouts/_header.html.erb]: Should only be shown to company admin

File: app/views/companies/show.html.erb [hunk 3]
```diff
@@ -25,5 +25,13 @@
...
 
+
+<% if current_user.invitations.include? @company %>
+  <p>
+    You have a pending invitation from this company. You can <%= link_to "Accept", "/user/invite/review/#{@company.id}/1" %> or <%= link_to "Decline", "/user/invite/review/#{@company.id}/0" %>
+  </p>
+<% end %>
+
+
 <%= link_to 'Edit', edit_company_path(@company) %> |
```
Comment (IvanPavlovic92) [inline app/views/companies/show.html.erb]: Should only be shown to company admin

**Comment by IvanPavlovic92** [comment]
Validations!

One of form fields should not be empty.
Email should only search for users with the email exact as input string.
First and Last name should have min 3 chars for search.

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -34,6 +34,104 @@ def destroy
...
+    not_in_company_users.each do |u|
+      if (u.invitations.include? current_user.current_company)
+        @users << {user: u, invitation: true}
+      else
+        @users << {user: u, invitation: false}
+      end
+    end
+
+    respond_to do |format|
+      if @users
```
Comment (IvanPavlovic92) [inline app/controllers/users_controller.rb]: change 
`if @users`
to
`if @users.any?`

**Comment by Baxxx** [comment]
@IvanPavlovic92 Shouldn't these validations be front-end? The `find` action can be reached only via post method.
Also, `Email should only search for users with the email exact as input string.` conflicts with the issue #16

**Comment by IvanPavlovic92** [comment]
Validations should be in both frontend and backend.
Email should pass email validations, need to have some chars than @ than some chars than . than chras.

so you could do that or simply set to be exactly same as email from database.

File: app/assets/javascripts/users.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+function search() {
```
Comment (sls-lnluksa) [inline app/assets/javascripts/users.js]: Change names of ```search``` and ```find``` to something more descriptive.
ex. searchUser

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -34,6 +34,132 @@ def destroy
...
+      else
+        format.json { render json: { message: "No users found." } }
+      end
+    end
+  end
+
+  def invite_to_company
+    authorize current_user
+    begin
+      if UserCompany.where("user_id = ? AND company_id = ?","#{params[:user_id]}","#{current_user.current_company.id}").any? # if there is an entry already
```
Comment (sls-lnluksa) [inline app/controllers/users_controller.rb]: Put this into variable

**Pull request merged**